src/gt4py/cartesian/definitions.py

```diff
@@ -101,6 +101,9 @@ class BuildOptions(AttributeClassLike):
     rebuild = attribute(of=bool, default=False)
     raise_if_not_cached = attribute(of=bool, default=False)
     cache_settings = attribute(of=DictOf[str, Any], factory=dict)
+    literal_precision = attribute(of=int, default=64)
+    "Specify the literal precision for automatic casts. Defaults to 64-bit"
```

Comment:
To make it immediately clear that `literal_precision` can either be 32 or 64, one might define the following type descriptor in `gt4py.cartesian.utils.attrib`:
```python3
def _make_literal_validator(args):
    if not isinstance(args, tuple):
        args = tuple([args])

    def _is_literal_validator(instance, attribute, value):
        if value not in args:
            raise ValueError(
                f"Invalid literal value {value} for attribute `{attribute.name}`; "
                f"expected one of {args}."
            )

    return _is_literal_validator


class _LiteralDescriptor:
    def __getitem__(self, values):
        return _TypeDescriptor(
            "Literal",
            args=values,
            make_validator_func=_make_literal_validator,
            type_hint=typing.Literal,
        )


Literal = _LiteralDescriptor()
```
so that here we could have:
```suggestion
    literal_precision = attribute(of=Literal[32, 64], default=64)
    "Specify the literal precision for automatic casts. Defaults to 64-bit"
```
Just thinking out loud, this comment should not be blocking.